app/controllers/mass_rollouts_controller.rb

```diff
@@ -16,23 +16,34 @@ def create
   end
 
   def deploy
-    environment = deploy_group.environment
-    template_stages = environment.template_stages.all
-    missing_only = params[:missing_only] == "true"
-    stages_to_deploy = missing_only ? deploy_group.stages.reject(&:last_successful_deploy) : deploy_group.stages
-    deploys = stages_to_deploy.map do |stage|
-      template_stage = template_stages.detect { |ts| ts.project_id == stage.project.id }
-      next unless template_stage
+    stages_to_deploy = deploy_group.stages
 
-      last_success_deploy = template_stage.last_successful_deploy
-      next unless last_success_deploy
+    if params[:status] == 'successful'
+      stages_to_deploy.select!(&:last_successful_deploy)
+    elsif params[:status] == 'missing'
+      stages_to_deploy.reject(&:last_successful_deploy)
+    end
 
-      deploy_service = DeployService.new(current_user)
-      deploy_service.deploy(stage, reference: last_success_deploy.reference)
+    if defined?(SamsonKubernetes::Engine)
+      if params[:type] == 'kubernetes'
+        stages_to_deploy.select!(&:kubernetes?)
+      elsif params[:type] == 'non_kubernetes'
+        stages_to_deploy.reject!(&:kubernetes?)
+      end
+    end
```

Comment:
prefer a single select per param like this (to avoid duplication and bugs)

```
if kubernetes = params[:kubernetes].presence
  stages_to_deploy.select! { |s| s.kubernetes?.to_s == kubernetes }
end
```